mobly/controllers/android_device.py

```diff
@@ -207,8 +207,15 @@ def get_instances_with_configs(configs):
             raise Error(
                 "Required value 'serial' is missing in AndroidDevice config %s."
                 % c)
-        ad = AndroidDevice(serial)
-        ad.load_config(c)
+        is_required = c.pop("required", True)
+        try:
+            ad = AndroidDevice(serial)
+            ad.load_config(c)
+        except:
+            if not is_required:
```

Comment:
Does this take into account other issues with the device, e.g. not responsive to adb/cannot connect to sl4a?